### PR TITLE
Update honnef.co/go/tools, unpin golang.org/x/sys

### PR DESCRIPTION
**Pull request by tamird**
Includes a small drive-by cleanup in storage.

### CODE REVIEW

File: build/style_test.go [hunk 1]
```diff
@@ -1015,15 +1015,15 @@ func TestStyle(t *testing.T) {
 				// which results in the unused warning:
 				//
 				// sql/parser/yaccpar:362:3: this value of sqlDollar is never used (SA4006)
-				&lint.GlobIgnore{Pattern: "github.com/cockroachdb/cockroach/pkg/sql/parser/yaccpar", Checks: []string{"SA4006"}},
+				&lint.GlobIgnore{Pattern: "github.com/cockroachdb/cockroach/pkg/sql/parser/sql.go", Checks: []string{"SA4006"}},
```
Comment (tschottdorf) [inline build/style_test.go]: Why this change? Do the comments need an update? Ditto below.

File: pkg/storage/allocator.go [hunk 2]
```diff
@@ -534,11 +534,8 @@ func (a Allocator) RebalanceTarget(
 			ReplicaID: rangeInfo.Desc.NextReplicaID,
 		}
 
-		// Deep-copy the Replicas slice since we'll mutate it.
 		desc := *rangeInfo.Desc
-		desc.Replicas = append([]roachpb.ReplicaDescriptor(nil), desc.Replicas...)
-
-		desc.Replicas = append(desc.Replicas, newReplica)
+		desc.Replicas = append(desc.Replicas[:len(desc.Replicas):len(desc.Replicas)], newReplica)
```
Comment (tschottdorf) [inline pkg/storage/allocator.go]: Please restore the comment. This code is nice and compressed but pretty sure most people will need to squint at it for a minute or two to figure out what's going on.

**Review by tbg — APPROVED** [review]
LGTM with some comments and also @bdarnell to approve the dep bumps.

File: build/style_test.go (hunk 1 above)
Comment (tamird) [reply build/style_test.go]: This is just reverting a previous change that was made necessary by an upstairs bug (see the comment being removed from Gopkg.toml)

File: pkg/storage/allocator.go (hunk 2 above)
Comment (tamird) [reply pkg/storage/allocator.go]: Restored.

File: build/style_test.go (hunk 1 above)
Comment (tamird) [reply build/style_test.go]: Er, I misspoke - there's no comment referring to the upstream bug, but here's where it got fixed: https://github.com/dominikh/go-tools/commit/80dfe5569e5c609daca40943b2385b4c6cc33b48

**Comment by bdarnell** [comment]
:lgtm:

---

Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/19691#-:-KxuLTjgVa4rNEv8cJqd:bnfp4nl)*
